ext/semian/resource.c

```diff
@@ -63,6 +85,10 @@ semian_resource_acquire(int argc, VALUE *argv, VALUE self)
     }
   }
 
+  // Restore the timeout values, in case they were mutated
+  res.timeout.tv_sec = tv_sec;
```

Comment:
In most cases this will be a noop, but we need this to potentially restore the backed-up value above.